pkg/lang/ir/conda.go

```diff
@@ -60,24 +62,32 @@ func (g Graph) compileCondaPackages(root llb.State) llb.State {
 		0755, llb.WithParents(true), llb.WithUIDGID(g.uid, g.gid)),
 		llb.WithCustomName("[internal] setting conda cache mount permissions"))
 
+	root = llb.User("envd")(root)
 	// Compose the package install command.
 	var sb strings.Builder
-	if len(g.CondaConfig.AdditionalChannels) == 0 {
-		sb.WriteString("/opt/conda/bin/conda install -n envd")
-
+	logrus.Debugf("conda packages: %s", g.CondaConfig.CondaPackages)
+	logrus.Debugf("conda env file content: %s", g.CondaConfig.CondaEnvFileContent)
+	if len(g.CondaConfig.CondaEnvFileContent) != 0 {
+		logrus.Debugf("using custom conda environment file content: %s", g.CondaConfig.CondaEnvFileContent)
+		root = root.File(llb.Mkfile(condaTempFile, 0644,
+			g.CondaConfig.CondaEnvFileContent, llb.WithUIDGID(g.uid, g.gid)),
+			llb.WithCustomName("[internal] create conda env file"))
+		sb.WriteString(fmt.Sprintf("/opt/conda/bin/conda env update -n envd --file %s", condaTempFile))
```

Comment:
Should we use the env file when init conda envd env?

In case user specified a Python version in the env file.